### PR TITLE
baremetal: make provisioningNetworkInterface optional

### PR DESCRIPTION
**Pull request by asalkeld**

### CODE REVIEW

**Comment by asalkeld** [comment]
/hold 
requires this to merge https://github.com/openshift/cluster-baremetal-operator/pull/132

**Comment by asalkeld** [comment]
/hold cancel
/retest
(dependent PR merged)

File: docs/user/metal/install_ipi.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ accessible over the external network which may not be desirable.
 
 * **NIC #2 - Provisioning Network (optional) **
   * A private network used for PXE based provisioning.
-  * You must specify `provisioningNetworkInterface` to indicate which
+  * You can specify `provisioningNetworkInterface` to indicate which
```
Comment (hardys) [inline docs/user/metal/install_ipi.md:53]: Should we mention that if not specified the interface is derived from the `bootMACAddress`?

**Comment by hardys** [comment]
This also depends on https://github.com/openshift/cluster-baremetal-operator/pull/149

**Comment by hardys** [comment]
Ugh, the manifests file isn't covered by baremetal-approvers OWNERS, but 

/lgtm

**Review by sadasu — COMMENTED** [review]
/lgtm

File: docs/user/metal/install_ipi.md (hunk 1 above)
Comment (sadasu) [reply docs/user/metal/install_ipi.md:53]: Should we also mention that when provisioningNetwrokInterface is specified, the user needs to make sure all the control plane hosts have an interface of that name?

**Comment by stbenjam** [comment]
/label platform/baremetal

Would you mind changing metal3 to baremetal in the commit and PR title? Metal3 doesn't mean much to the installer folks.

**Comment by stbenjam** [comment]
- Can you add `+optional` here?  https://github.com/openshift/installer/blob/2e88addc34c0020ee060ca270d45389095450ee5/pkg/types/baremetal/platform.go#L118-L120
- You'll need to re-run verify-codegen.sh after that

File: docs/user/metal/install_ipi.md (hunk 1 above)
Comment (asalkeld) [reply docs/user/metal/install_ipi.md:53]: thanks, updated. Apologies for the slow response.

**Comment by patrickdillon** [comment]
/approve

Also, please include more informative commit messages according to the [contributing guidelines](https://github.com/openshift/installer/blob/master/CONTRIBUTING.md#commit-message-format).

**Comment by hardys** [comment]
@patrickdillon @sadasu @stbenjam FYI this caused a regression ref https://bugzilla.redhat.com/show_bug.cgi?id=1984576

Looking at the test history, I can see that pull-ci-openshift-installer-master-e2e-metal-ipi-ovn-ipv6 never passed, we need to be careful to ensure that passes before merging any baremetal IPI PRs

**Comment by hardys** [comment]
https://github.com/openshift/installer/pull/5100 should resolve the regression I think.

**Comment by stbenjam** [comment]
> Looking at the test history, I can see that pull-ci-openshift-installer-master-e2e-metal-ipi-ovn-ipv6 never passed, we need to be careful to ensure that passes before merging any baremetal IPI PRs

Yea sorry I missed this.  I think the installer team normally relies on the platform teams to /lgtm before merging and take that as our word we think the change is good.

**Pull request merged**